Jenkinsfile

```diff
@@ -1182,6 +1185,19 @@ pipeline {
                             run_if_landing: false,
                             job_status: job_status_internal
                         ),
+                        'Functional Hardware Medium VMD': getFunctionalTestStage(
+                            name: 'Functional Hardware Medium VMD',
+                            pragma_suffix: '-hw-medium-vmd',
+                            label: params.FUNCTIONAL_HARDWARE_MEDIUM_VMD_LABEL,
+                            next_version: next_version,
+                            stage_tags: 'hw_vmd,medium,-provider',
```

Comment:
Do we need `-provider`? This implies we would need yet another stage
```
Functional Hardware Medium VMD Provider
```

Reply:
The `provider` tag is used to identify tests that should be run against different providers - currently the daos_test suite of tests, so `-provider` indirectly indicates a stage that will be run with the best provider supported by the selected cluster.

The `hw_vmd` tag will already exclude current group of tests using the `provider` tag as they also use the `hw`, so leaving or removing the `-provider` tag currently has no impact.  

We would only ever need a `Functional Hardware Medium <provider> Provider VMD` stage if we needed to ensure we ran the provider tagged tests (e.g. daos_test) with VMD. Right now the NVMe/VMD testing is luck of the draw based upon the cluster assignment (like other Medium stages).

Reply:
I'm not going to block either way, but I do think it's better to omit it to keep things simple since we don't have a VMD provider stage. But yeah, we don't have `hw_vmd,provider` tests anyway